### ISSUE TITLE
Installer should skip packages when a newer version is already installed

### ISSUE THREAD

**Issue by ohine** [dnnsoftware/Dnn.Platform#2458]
## Description of problem
When upgrading Dnn, the process will fail if you have a newer version of an extension already installed which is also included in the platform upgrade package. We have run into this with sites when they upgrade to v8.0.4 due to the SecurityAnalyzer and version mismatches.

## Description of solution
We should modify the upgrader to skip any packages which a newer version is already installed. 

## Affected version

* [x] All Versions

**Comment by sleupold** [comment]
IMO we should detect just the latest version in that folder and delete all other packages prior to install (like the currently installed one is deleted after the install)

**Comment by ohine** [comment]
@sleupold that wouldn’t solve the issue. The problem isn’t we have multiple versions staged to install. The problem is we have a newer version already installed and dnn isn’t prepared for such a situation and errors out.

**Comment by sleupold** [comment]
AFAIK the installer terminates with an error, when a newer version of the same module is already installed. This used to be in place since DNN 4

**Comment by ohine** [comment]
That’s the problem. It errors out when it could easily continue.

**Comment by sleupold** [comment]
ah, I see, it should log the issue and continue with the next package.

**Comment by Accuraty** [comment]
Just to confirm the current behavior. If you have DNN 7.04.02 with Security Analyzer 8.1.3 already installed, when you upgrade to 8.00.04 it errors out because Security Analyzer (v1.0.2 I think?) is in the Install/Modules folder. Further... if you delete the Security Analyzer module package from that folder and then attempt to restart/resume the DNN upgrade install, it finishes successfully (or at least appears to do so). So I agree, log/report the error and continue sounds great to me! Especially since a user new to upgrading DNN would have almost no idea how to work through this error and stop leaving a seemingly incomplete upgrade.

**Comment by valadas** [comment]
Yes, I think it should just log as Info and continue with next package.

**Comment by valadas** [comment]
I am closing this issue since it had no progress. If someone plans on submitting a PR for that any soon, please comment or reopen

**Comment by mitchelsellers** [comment]
Reopening this one as it is a needed item to provide for a safer upgrade experience going forward.
